src/commands/shane/data/file/download.ts

```diff
@@ -0,0 +1,93 @@
+
+    public static examples = [
+        `sfdx shane:data:file:download -d 0691k000000MXfkAAG -o ./files/
+    //save a ContentDocument from the org to the files directory, keeping the existing filename`,
+        `sfdx shane:data:file:download -d 0691k000000MXfkAAG -o ./files/King.jpg
+    //save a ContentDocument from the org to files/King.jpg`,
+        `sfdx shane:data:file:download -d 0691k000000MXfkAAG
+    //save a ContentDocument from the org to the current working directory, keeping the existing filename`,
+        `sfdx shane:data:file:download -v 0681k000000MXfkAAG -o ./files/King.jpg
+    //save a ContentVersion from the org to files/King.jpg`
+    ];
+
+    protected static flagsConfig = {
+        documentid: flags.id({ char: 'd', description: 'optional ContentDocument ID that should be downloaded' }),
+        documentversionid: flags.id({ char: 'v', description: 'optional ContentVersion ID that should be downloaded' }),
+        outputpath: flags.filepath({
+            char: 'o',
+            description:
+                'optional output path to save the file, if ommitted will save to current directory, ' +
+                'if directory then it will keep the filename and save into that directory'
+        })
+    };
+
+    // Comment this out if your command does not require an org username
+    protected static requiresUsername = true;
+
+    // tslint:disable-next-line:no-any
+    public async run(): Promise<any> {
+        return new Promise(async (resolve, reject) => {
+            if (this.flags.documentid && this.flags.documentversionid) {
+                reject('Please include either documentid or documentVerionId but not both');
+            }
+            if (!this.flags.documentid && !this.flags.documentversionid) {
+                reject('Please include one of documentid or documentVerionId');
+            }
+
+            // this.org is guaranteed because requiresUsername=true, as opposed to supportsUsername
+            const conn = this.org.getConnection();
+            // tslint:disable-next-line:prefer-const
+            let versionId = this.flags.documentversionid;
+            if (this.flags.documentid) {
+                versionId = (<ContentDocument>await conn.sobject('ContentDocument').retrieve(this.flags.documentid)).LatestPublishedVersionId;
+            }
+
+            const version = <ContentVersion>await conn.sobject('ContentVersion').retrieve(versionId);
+
+            // Output starts are the files stored name and extention
+            // tslint:disable-next-line:prefer-const
+            let targetFilename = `${version.Title}.${version.FileExtension}`;
+            // If output path is specified then assign it
+            if (this.flags.outputpath) {
+                targetFilename = this.flags.outputpath;
+            }
+            // Try and test the path to see if it is a directory, if it is then merge it with the stored name
+            fs.stat(targetFilename, async (err, fStat) => {
```

Comment:
maybe fs.ensureDir ?

Reply:
The reason I hadn't tackled this is because I was being lazy and didn't want to determine if the thing they gave me was a directory or a filename. Need to decide if we're going to assume that anything with a `.` in the final section of the path is a filename and can be stripped off and then the remainder passed to ensureDir.